src/topology/vector_bundle.lean

```diff
@@ -0,0 +1,216 @@
+work with vector bundles this way.
+
+## Definitions
+
+* `nc_topological_vector_bundle R E F`  : topological vector bundle with non constant fiber. It is
+                                          implemented as a `class : Prop`. Here `F` is a function
+                                          `I → Type*` and `I` is an index type.
+* `topological_vector_bundle R E F`     : topological vector bundle constant fiber `F : Type*`. It
+                                          also implemented as a `class : Prop`,
+
+-/
+
+noncomputable theory
+
+variables {B : Type*}
+
+/--
+`total_space E` is the total space of the bundle `Σ x, E x`. This type synonym is used to avoid
+conflicts with general sigma types.
+-/
+def bundle.total_space (E : B → Type*) := Σ x, E x
+
+instance {E : B → Type*} [inhabited B] [inhabited (E (default B))] :
+  inhabited (bundle.total_space E) := ⟨⟨default B, default (E (default B))⟩⟩
+
+/-- `bundle.proj E` is the canonical projection `total_space E → B` on the base space. -/
+def bundle.proj (E : B → Type*) : bundle.total_space E → B := λ y : (bundle.total_space E), y.1
+
+open bundle
+
+variables (R : Type*) (E : B → Type*) (F : Type*)
+[comm_semiring R] [∀ x, add_comm_monoid (E x)] [∀ x, semimodule R (E x)]
+[∀ x, topological_space (E x)]
+
+/-- `bundle.dual R E` is the dual bundle. -/
+@[reducible] def bundle.dual := (λ x, module.dual R (E x))
+
+localized "notation E `ᵛ` R := bundle.dual R E" in bundle.dual
+
+section
+
+variables [topological_space B] [topological_space F] [topological_space (total_space E)]
+[add_comm_monoid F] [semimodule R F]
+
+@[nolint unused_arguments]
+instance {x : B} : has_coe_t (E x) (total_space E) := ⟨λ y, (⟨x, y⟩ : total_space E)⟩
+
+lemma to_total_space_coe {x : B} {v : E x} : (v : total_space E) = ⟨x, v⟩ := rfl
+
+/-- Local trivialization for vector bunlde. -/
+@[nolint has_inhabited_instance]
+structure vector_bundle_trivialization extends bundle_trivialization F (proj E) :=
+(linear : ∀ x ∈ base_set, is_linear_map R (λ y : (E x), (to_fun y).2))
+
+instance : has_coe (vector_bundle_trivialization R E F) (bundle_trivialization F (proj E)) :=
+⟨vector_bundle_trivialization.to_bundle_trivialization⟩
+
+instance : has_coe_to_fun (vector_bundle_trivialization R E F) :=
+⟨_, λ e, e.to_bundle_trivialization⟩
+
+@[simp] lemma coe_eq_coe_coe {e : vector_bundle_trivialization R E F} :
+  (⇑e : (total_space E) → B × F) =
+  ((e : bundle_trivialization F (proj E)) : (total_space E) → B × F) :=
+rfl
+
+section
+
+def vector_bundle_trivialization.at (e : vector_bundle_trivialization R E F) (b : B)
+  (hb : b ∈ e.base_set):
+  continuous_linear_equiv R (E b) F :=
+{ to_fun := λ y, (e.to_fun y).2,
+  inv_fun := λ z, begin let g := (e.to_local_homeomorph.symm ⟨b, z⟩).2,
+    have h : ((e.to_bundle_trivialization.to_local_homeomorph.symm) (b, z)).fst = b := sorry,
+    rw h at g,
+    exact g,
+  end,
+  left_inv := begin
+  end, }
```

Comment:
Well this should make sense mathematically but I don't know if it makes sense in Lean: there is always the problem that things are equal but not definitionally equal and so the inverse function has to be defined in tactic mode and then I don't know anymore how to deal with it. If things cannot be done this way they probably should be done, as Sébastien was suggesting, by requiring stuff as part of the definition but it would be cool if things could be done this way. I need help with this as I do not at all know how to solve these problems and fill up the fields of this function.

Reply:
This is meant to address the comment

> There is still a problem with your definitions: in a topological vector bundle, the fibers should be topological vector spaces, and the trivializations in the fibers should be continuous linear maps with continuous linear inverse.